### ISSUE TITLE
Confusing use of `notify`

### ISSUE THREAD

**Issue by bbros-dev** [tokio-rs/mini-redis#86]
Thank you for all the effort put into Tokio and the mini-redis demonstration project. As well as for making these open source.
 
Trying to understand Rust, we were confused by the use of the term `notify` to refer to a `Receiver`, specifically [`notify.recv()`](https://github.com/tokio-rs/mini-redis/blob/master/src/shutdown.rs#L44) had us looking twice.

Some dictionaries have `listen` or `collect` or `gather` as antonyms for notify.

We think changing this usage of `notify` to one of the antonyms may be less puzzling?

**Comment by Darksonn** [comment]
Changing it to `listen` or `receiver` sounds reasonable to me. I do not think that `collect` and `gather` would be good names.

**Comment by bbros-dev** [comment]
Thanks for the feedback.  We had contemplated `receiver` but thought for new users, coming from other languages, it might be mistaken as an instance  (or some such concept) of `Receiver`, while from (weak/flawed) memory it is a `Future`.

We're still working through this example so will likely wait a while before making an PR - to ensure we haven't misunderstood.

For future reference, we also thought [this line](https://github.com/tokio-rs/mini-redis/blob/master/src/shutdown.rs#L5) (signalling is a synonym for sending):

```rust
/// Shutdown is signalled using a `broadcast::Receiver`.

```

might be better phrased:

```rust
/// Shutdown signals are monitored via a `broadcast::Receiver`.
```

**Comment by Darksonn** [comment]
> We had contemplated receiver but thought for new users, coming from other languages, it might be mistaken as an instance (or some such concept) of `Receiver`, while from (weak/flawed) memory it is a `Future`.

No, it isn't a `Future`. It is an instance of the type [`tokio::sync::broadcast::Receiver`][1].

---

As for the word "monitor", I find that word very unnatural here. If we are to rephrase the sentence, I would prefer this:
```
/// Shutdown signals are received using a `broadcast::Receiver`.
```


[1]: https://docs.rs/tokio/1/tokio/sync/broadcast/struct.Receiver.html

**Comment by bbros-dev** [comment]
`receiver` it is then.
